test/e2e/service.go

```diff
@@ -1022,60 +1024,83 @@ func validateUniqueOrFail(s []string) {
 	}
 }
 
-func getPortsByIp(subsets []api.EndpointSubset) map[string][]int {
-	m := make(map[string][]int)
-	for _, ss := range subsets {
+func getContainerPortsByPodUID(endpoints *api.Endpoints) PortsByPodUID {
+	m := PortsByPodUID{}
+	for _, ss := range endpoints.Subsets {
 		for _, port := range ss.Ports {
 			for _, addr := range ss.Addresses {
-				Logf("Found IP %v and port %v", addr.IP, port.Port)
-				if _, ok := m[addr.IP]; !ok {
-					m[addr.IP] = make([]int, 0)
+				containerPort := port.Port
+				hostPort := port.Port
+
+				// use endpoint annotations to recover the container port in a Mesos setup
+				// compare contrib/mesos/pkg/service/endpoints_controller.syncService
+				// TODO(sttts): add ContainerPort to EndpointPort struct, defaulting to (host) Port
```

Comment:
i'm not sure about this TODO, doesn't really seem like the right way to solve this. perhaps remove the TODO here and add an issue for tracking this idea in the k8s-mesos repo?